### ISSUE TITLE
Only allow SearchInput to use placeholder

### ISSUE THREAD

**Issue by J-Kallunki** [vrk-kpa/suomifi-ui-components#185]
1. Make sure that `SearchInput `-component `placeholder` is implemented the same way as `placeholder` in `TextInput`.
2. Remove the variant of `TextInput` component that uses `placeholder`.
3. Add `placeholder` attribute to `TextInput` -interface without description. Remove it from html output. Add `logger.error` that it is not accessible when using `placeholder`-prop/attribute and it is ignored.

https://www.smashingmagazine.com/2018/06/placeholder-attribute/

**Comment by LJKaski** [comment]
Couldn't make things work with the above workflow since SearchInput uses TextInput. Need to make some changes to the way these work. Doing it as follows:

1. Make a new class of BaseTextInput which essentially is the current TextInput without the required error when using placeholder.
2. Exporting another class by the name of TextInput, which extends BaseTextInput and adds the error message.
3. Set the SearchInput to extend the BaseTextInput class to allow the use of placeholder

The BaseTextInput is not listed in the Index.ts.
